### PR TITLE
Added cmake_modules to configure with ROS Indigo

### PR DESCRIPTION
**Pull request by stefangachter**
Change CMakeLists.txt and package.xml to be able to configure with ROS Indigo

### CODE REVIEW

File: aslam_offline_calibration/kalibr/python/kalibr_common/ImageDatasetReader.py [hunk 1]
```diff
@@ -114,6 +114,6 @@ def getImage(self,idx):
             img_data = np.reshape(self.uncompress(np.fromstring(data.data, dtype='uint8')),(data.height,data.width), order="C")
             
         else:
-            img_data = np.array(self.CVB.imgmsg_to_cv(data))  
+            img_data = np.squeeze(np.array(self.CVB.imgmsg_to_cv2(data, "mono8")))
```
Comment (NikolausDemmel) [inline aslam_offline_calibration/kalibr/python/kalibr_common/ImageDatasetReader.py]: Why did you add `squeeze` here? Is that strictly needed?

File: aslam_offline_calibration/kalibr/python/kalibr_camera_focus [hunk 2]
```diff
@@ -19,7 +19,7 @@ class CameraFocus:
     def callback(self, msg):
         #convert image to opencv
         try:
-            cv_image = self.bridge.imgmsg_to_cv(msg)
+            cv_image = self.bridge.imgmsg_to_cv2(msg, "mono8")
```
Comment (NikolausDemmel) [inline aslam_offline_calibration/kalibr/python/kalibr_camera_focus]: Does this addition mean it works also with color images as input?

**Comment by NikolausDemmel** [comment]
General remark: This PR adds the executable flag to several files, which it probably should not.

**Comment by NikolausDemmel** [comment]
Otherwise: lgtm for indigo.

File: Schweizer-Messer/sm_matrix_archive/package.xml [hunk 3]
```diff
@@ -7,7 +7,7 @@
   <maintainer email="paul.furgale@mavt.ethz.ch">Paul Furgale</maintainer>
   <license>New BSD</license>
   <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>cmake_modules</build_depend>
```
Comment (simonlynen) [inline Schweizer-Messer/sm_matrix_archive/package.xml]: add separator between build_tool and build depends

File: Schweizer-Messer/numpy_eigen/package.xml [hunk 4]
```diff
@@ -7,7 +7,7 @@
   <maintainer email="paul.furgale@mavt.ethz.ch">Paul Furgale</maintainer>
   <license>New BSD</license>
   <buildtool_depend>catkin</buildtool_depend>
-
```
Comment (simonlynen) [inline Schweizer-Messer/numpy_eigen/package.xml]: add separator between build_tool and build depend.

File: Schweizer-Messer/sm_eigen/package.xml [hunk 5]
```diff
@@ -7,11 +7,10 @@
   <maintainer email="paul.furgale@mavt.ethz.ch">Paul Furgale</maintainer>
   <license>New BSD</license>
   <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>cmake_modules</build_depend>
```
Comment (simonlynen) [inline Schweizer-Messer/sm_eigen/package.xml]: same

File: Schweizer-Messer/sm_kinematics/package.xml [hunk 6]
```diff
@@ -6,18 +6,15 @@
 
   <maintainer email="paul.furgale@mavt.ethz.ch">Paul Furgale</maintainer>
   <license>New BSD</license>
-  
   <buildtool_depend>catkin</buildtool_depend>
-
```
Comment (simonlynen) [inline Schweizer-Messer/sm_kinematics/package.xml]: same

File: Schweizer-Messer/sm_kinematics/package.xml [hunk 7]
```diff
@@ -6,18 +6,15 @@
 
   <maintainer email="paul.furgale@mavt.ethz.ch">Paul Furgale</maintainer>
   <license>New BSD</license>
-
```
Comment (simonlynen) [inline Schweizer-Messer/sm_kinematics/package.xml]: I think these separators can be kept for readability.

File: Schweizer-Messer/sm_kinematics/package.xml [hunk 8]
```diff
@@ -6,18 +6,15 @@
...
   <license>New BSD</license>
-  
   <buildtool_depend>catkin</buildtool_depend>
-  
+  <build_depend>cmake_modules</build_depend>
   <build_depend>sm_boost</build_depend>
   <build_depend>sm_common</build_depend>
   <build_depend>sm_eigen</build_depend>
   <build_depend>sm_random</build_depend>
-
```
Comment (simonlynen) [inline Schweizer-Messer/sm_kinematics/package.xml]: also here.

File: aslam_offline_calibration/ethz_apriltag2/package.xml [hunk 9]
```diff
@@ -1,17 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <package>
   <name>ethz_apriltag2</name>
-  <description>
-  ethz_apriltag2
-
-	ros lib of http://people.csail.mit.edu/kaess/apriltags/
```
Comment (simonlynen) [inline aslam_offline_calibration/ethz_apriltag2/package.xml]: Why this change?

**Comment by simonlynen** [comment]
@stefangachter a few remarks. To keep the CL focused please only add the changes for the CMakeModules dependency here. The changes in the python script should be part of a different change-list.

Also please revert the file-mode changes as well as the formatting of the package.xml.

Otherwise this looks good. Thanks for the fix.
